internal/impl/snowflake/output_snowflake_put.go

```diff
@@ -670,6 +672,12 @@ func (s *snowflakeWriter) callSnowpipe(ctx context.Context, snowpipe, requestID,
 }
 
 func (s *snowflakeWriter) WriteBatch(ctx context.Context, batch service.MessageBatch) error {
+	s.connMut.Lock()
+	defer s.connMut.Unlock()
```

Comment:
I'm not sure how to tell if / when this mutex is necessary. For example, the `kafka_franz` output doesn't have it, but the `nats_jetstream` one [does](https://github.com/benthosdev/benthos/blob/63559aa93c1e00632ba5e2448c3b9b97f552ad64/internal/impl/nats/output_jetstream.go#L74).

Reply:
It's probably not needed, a mutex around connections is just an extra blanket of safety as it's the role of the channel based output wrapper to only call Close on a single thread. It would however be a problem if under certain circumstances we were to disconnect the client from the write call, as write can be called in parallel (as can connect).

Overall though it's never a problem to add a mutex just in case.